lib/fetchHttpClient.ts

```diff
@@ -0,0 +1,210 @@
+            }
+          }
+
+          httpRequest.body = requestForm;
+          httpRequest.formData = undefined;
+          const contentType = httpRequest.headers.get("Content-Type");
+          if (contentType && contentType.indexOf("multipart/form-data") !== -1) {
+            if (typeof requestForm.getBoundary === "function") {
+              httpRequest.headers.set("Content-Type", `multipart/form-data; boundary=${requestForm.getBoundary()}`);
+            } else {
+              // browser will automatically apply a suitable content-type header
+              httpRequest.headers.remove("Content-Type");
+            }
+          }
+        }
+
+        let body = httpRequest.body
+            ? (typeof httpRequest.body === "function" ? httpRequest.body() : httpRequest.body)
+            : undefined;
+        if (httpRequest.onUploadProgress && httpRequest.body) {
+            let loadedBytes = 0;
+            const uploadReportStream = new Transform({
+                transform: (chunk: string | Buffer, _encoding, callback) => {
+                    loadedBytes += chunk.length;
+                    httpRequest.onUploadProgress!({ loadedBytes });
+                    callback(undefined, chunk);
+                }
+            });
+
+            if (isReadableStream(body)) {
+                body.pipe(uploadReportStream);
+              } else {
+                uploadReportStream.end(body);
+              }
+
+            body = uploadReportStream;
+        }
+
+        if (this.cookieJar && !httpRequest.headers.get("Cookie")) {
+            const cookieString = await new Promise<string>((resolve, reject) => {
+              this.cookieJar!.getCookieString(httpRequest.url, (err, cookie) => {
+                if (err) {
+                  reject(err);
+                } else {
+                  resolve(cookie);
+                }
+              });
+            });
+
+            httpRequest.headers.set("Cookie", cookieString);
+        }
+
+        const requestInit: RequestInit = {
+            body: body,
+            headers: httpRequest.headers.rawHeaders(),
+            method: httpRequest.method,
+            signal: abortController.signal
+        };
+
+        try {
+            const response: Response = await fetch(httpRequest.url, requestInit);
+
+            const headers = parseHeaders(response.headers);
+            const operationResponse: HttpOperationResponse = {
+                headers: headers,
+                request: httpRequest,
+                status: response.status
+            };
+
+            const onDownloadProgress = httpRequest.onDownloadProgress;
+            if (!onDownloadProgress) {
+                const bodyAsText = response.body ? await response.text() : undefined;
+                operationResponse.bodyAsText = bodyAsText;
+            } else {
+                const responseBody: ReadableStream<Uint8Array> | undefined = response.body ? response.body : undefined;
```

Comment:
Does this ReadableStream come from fetch spec? Is it actually implemented in node now?

Reply:
Maybe consider `response.body || undefined` if the goal is to normalize all falsy values to undefined